rosbag2_compression/test/rosbag2_compression/test_zstd_compressor.cpp

```diff
@@ -81,3 +101,68 @@ TEST_F(CompressionHelperFixture, zstd_compress_file_uri)
   EXPECT_GT(compressed_file_size, 0u);
   EXPECT_TRUE(rosbag2_storage::FilesystemHelper::file_exists(compressed_uri));
 }
+
+TEST_F(CompressionHelperFixture, zstd_decompress_file_uri)
+{
+  const auto uri = rosbag2_storage::FilesystemHelper::concat({temporary_dir_path_, "file1.txt"});
+  create_garbage_file(uri);
+  const auto initial_file_size = rosbag2_storage::FilesystemHelper::get_file_size(uri);
+
+  auto zstd_compressor = rosbag2_compression::ZstdCompressor();
+  const auto compressed_uri = zstd_compressor.compress_uri(uri);
+
+  ASSERT_EQ(0, std::remove(uri.c_str()));  // The test is invalid if the initial file is not deleted
+
+  auto zstd_decompressor = rosbag2_compression::ZstdDecompressor();
+  const auto decompressed_uri = zstd_decompressor.decompress_uri(compressed_uri);
+
+  const auto expected_decompressed_uri = uri;
+  const auto decompressed_file_size =
+    rosbag2_storage::FilesystemHelper::get_file_size(decompressed_uri);
+
+  EXPECT_NE(compressed_uri, uri);
+  EXPECT_NE(decompressed_uri, compressed_uri);
+  EXPECT_EQ(uri, expected_decompressed_uri);
+  EXPECT_EQ(initial_file_size, decompressed_file_size);
+  EXPECT_TRUE(rosbag2_storage::FilesystemHelper::file_exists(decompressed_uri));
+}
+
+TEST_F(CompressionHelperFixture, zstd_decompress_file_contents)
+{
+  const auto uri = rosbag2_storage::FilesystemHelper::concat({temporary_dir_path_, "file2.txt"});
+  create_garbage_file(uri);
+
+  auto compressor = rosbag2_compression::ZstdCompressor{};
+  const auto compressed_uri = compressor.compress_uri(uri);
+
+  auto decompressor = rosbag2_compression::ZstdDecompressor{};
+  const auto decompressed_uri = decompressor.decompress_uri(compressed_uri);
+
+  const auto initial_data = read_file(uri);
+  const auto initial_file_size = rosbag2_storage::FilesystemHelper::get_file_size(uri);
+  const auto decompressed_data = read_file(decompressed_uri);
+  const auto decompressed_file_size =
+    rosbag2_storage::FilesystemHelper::get_file_size(decompressed_uri);
+
+  EXPECT_EQ(initial_data.size() * sizeof(char), initial_file_size);
+  EXPECT_EQ(decompressed_data.size() * sizeof(char), decompressed_file_size);
```

Comment:
Can
```
sizeof(decltype(initial_data)::value_type)
sizeof(decltype(decompressed_data)::value_type)
```
be used here?

Btw, `sizeof(char) == 1` always (http://www.cs.technion.ac.il/users/yechiel/c++-faq/sizeof-char.html), even if `CHAR_BITS > 8` (https://stackoverflow.com/questions/21080114/if-char-bit-8-what-is-the-size-of-other-types).

Reply:
Interesting, one of the reasons why `vector<uint8_t>` is used over `vector<char>` is to handle systems when "`char` is not a byte".

Reply:
If I understand correctly, `char` is always a byte, and if a byte is more than 8 bits on a particular system then `uint8_t` will not be available on that system (your `vector<uint8_t>` won't compile). Let's not worry about that right now.

Reply:
Yeah, that's all for a different PR.